### ISSUE TITLE
RH nfd-operator excessively logs at over 15,000 logged lines per minute

### ISSUE THREAD

**Issue by worsco** [openshift/cluster-nfd-operator#133]
After installing the RH nfd-operator and gpu-operator in OCP 4.5 cluster, the nfd-operator logs at a rate of over 15,000 lines per minute.

```bash
$ oc get csv nfd.4.5.0-202011132127.p0
NAME                        DISPLAY                  VERSION   REPLACES   PHASE
nfd.4.5.0-202011132127.p0   Node Feature Discovery   4.5.0                Succeeded
```

```bash
$ date ; oc logs nfd-operator-7c69cf4f44-jkbwq | wc -l
Wed, Feb  3, 2021  2:44:30 PM
171957

$ date ; oc logs nfd-operator-7c69cf4f44-jkbwq | wc -l
Wed, Feb  3, 2021  2:47:01 PM
203070
```

2 minutes, 31 seconds since last count

171,975 - 203,070 = 31,095 messages

31095 logs / 2.5 minutes = 15547 logs

Typical contents of logs:

```
{"level":"info","ts":1612381861.9975307,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","Service":"nfd-master","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.0078075,"logger":"controller_nodefeaturediscovery","msg":"Found, skpping update","ServiceAccount":"nfd-worker","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.0079691,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","Role":"nfd-worker","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.01564,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","RoleBinding":"nfd-worker","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.0217285,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","ConfigMap":"nfd-worker","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.0276768,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","SecurityContextConstraints":"nfd-worker","Namespace":"default"}
{"level":"info","ts":1612381862.0387275,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","DaemonSet":"nfd-worker","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.0495317,"logger":"controller_nodefeaturediscovery","msg":"Looking for","ServiceAccount":"nfd-master","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.049615,"logger":"controller_nodefeaturediscovery","msg":"Found, skpping update","ServiceAccount":"nfd-master","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.04964,"logger":"controller_nodefeaturediscovery","msg":"Looking for","ClusterRole":"nfd-master","Namespace":""}
{"level":"info","ts":1612381862.04966,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","ClusterRole":"nfd-master","Namespace":""}
{"level":"info","ts":1612381862.0553772,"logger":"controller_nodefeaturediscovery","msg":"Looking for","ClusterRoleBinding":"nfd-master","Namespace":""}
{"level":"info","ts":1612381862.0554087,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","ClusterRoleBinding":"nfd-master","Namespace":""}
{"level":"info","ts":1612381862.0618849,"logger":"controller_nodefeaturediscovery","msg":"Looking for","DaemonSet":"nfd-master","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.0619338,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","DaemonSet":"nfd-master","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.1010854,"logger":"controller_nodefeaturediscovery","msg":"Looking for","Service":"nfd-master","Namespace":"openshift-operators"}
{"level":"info","ts":1612381862.1012244,"logger":"controller_nodefeaturediscovery","msg":"Found, updating","Service":"nfd-master","Namespace":"openshift-operators"}
```

**Comment by worsco** [comment]
Perhaps we're using the newest Operator SDK?  I believe, unless you tell it otherwise, it will run in logging mode set to debug.  I know of no way of configuring the operator, a custom resource, to change the logging level of the operator pod.

**Comment by worsco** [comment]
Appears the operator is logging every reconciliation request in https://github.com/openshift/cluster-nfd-operator/blob/924591e0cc495a649b885d5b629172d31098c2d7/pkg/controller/nodefeaturediscovery/nodefeaturediscovery_controls.go#L118

**Comment by ArangoGutierrez** [comment]
-> @zvonkok

**Comment by ArangoGutierrez** [comment]
fixed at [release-4.5] Bug 1940205: nodefeaturediscovery_state: avoid accumulating states #143 
/close
